File: 581-shortest-unsorted-continuous-subarray/581-shortest-unsorted-continuous-subarray.java
class Solution {
    public int findUnsortedSubarray(int[] nums) {
        int [] nums2 = new int[nums.length];
        System.arraycopy(nums,0,nums2,0,nums.length);
        Arrays.sort(nums2);
        
        int start =0, end =0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]!=nums2[i]){
                start = i; break;
            }
        }
        
        for(int i=nums.length-1;i>=0;i--){
            if(nums[i]!=nums2[i]){
                end = i; break;
            }
        }
        
        return (end-start==0? 0 : (end-start)+1);
    }
}

File: 581-shortest-unsorted-continuous-subarray/581-shortest-unsorted-continuous-subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        vector<int> v(nums);
        sort(v.begin(),v.end());
        int start =0 , end =0;
        for(int i=0;i<v.size();i++){
            if(v[i]!=nums[i]){
                start = i; break;
            }
        }
        
        for(int i=v.size()-1;i>=0;i--){
            if(v[i]!=nums[i]){
                end = i ; break;
            }
        }
        
        return (end-start==0? 0 : (end-start)+1);
        
    }
};